sdk/python/feast/infra/provider.py

```diff
@@ -61,6 +63,18 @@ def update_infra(
         """
         ...
 
+    def plan_infra(
+        self, config: RepoConfig, desired_registry_proto: RegistryProto
+    ) -> Infra:
+        """
+        Returns the Infra required to support the desired registry.
+
+        Args:
+            config: The RepoConfig for the current FeatureStore.
+            desired_registry_proto: The desired registry, in proto form.
+        """
+        return Infra()
```

Comment:
does it make sense to print a message or something here? e.g. so a custom provider knows that feast plan won't cover their use case? 

Not sure how we intend the SDK version of plan to be used, so maybe it doesn't make sense to log though.